src/app.ts

```diff
@@ -95,6 +98,15 @@ export class App extends Construct {
     this.yamlOutputType = props.yamlOutputType ?? YamlOutputType.FILE_PER_CHART;
   }
 
+  /**Sets a new IConstruct and a new DependencyGraph for the object */
+  private commonFunc(): void {
```

Comment:
1. I can't seem any use of "found" neither in the new method nor in the original `synth()`. I think we can just remove it.
2. As for `charts`, I wouldn't store it as a private field. Instead, I'd just implement it like so:

```ts
/**
 * Returns all the charts in this app, sorted topologically.
 */
 public get charts(): Chart[] {
  const isChart = (x: IConstruct): x is Chart => x instanceof Chart;
  return new DependencyGraph(Node.of(this))
    .topology()
    .filter(isChart)
}
```

Notice that this now returns `Chart[]`, so `Chart.of(node)` is not needed anymore

Reply:
Made the necessary changes. This looks a lot cleaner.